app/controllers/user.js

```diff
@@ -67,3 +67,20 @@ exports.signUp = (req, res, next) => {
       next(error);
     });
 };
+
+exports.listUsers = (req, res, next) => {
+  const auth = req.headers.authorization; // auth is in base64(username:password)  so we need to decode the base64
+  try {
+    const decoded = sessionsManager.decode(auth);
+  } catch (e) {
+    throw errors.unauthorizedNoLogin;
+  }
```

Comment:
lets move this auth logic to a middleware, so we can call it in the route itself an re use it in any other auth endpoint.